core/src/main/java/tc/oc/pgm/flag/FlagParser.java

```diff
@@ -50,29 +56,81 @@ private void checkDeprecatedFilter(Element el) throws InvalidXMLException {
     }
   }
 
-  public Post parsePost(Element el) throws InvalidXMLException {
+  private void checkDeprecatedMultiPost(Element el) throws InvalidXMLException {
+    if (el.getChildren("post").size() <= 1) return;
+    throw new InvalidXMLException(
+        "Multiple 'post' elements inside 'flag' are no longer supported, use a single 'post' with multiple inner 'post' elements instead.\n"
+            + "Check the docs at https://pgm.dev or PR #984 on github for more details.",
+        el);
+  }
+
+  public PostDefinition parsePost(Element el) throws InvalidXMLException {
     checkDeprecatedFilter(el);
 
-    @Nullable String name = el.getAttributeValue("name");
+    PostDefinition post =
+        el.getChildren("post").isEmpty() ? parseSinglePost(el) : parseCompositePost(el);
+
+    posts.add(post);
+    factory.getFeatures().addFeature(el, post);
+
+    return post;
+  }
+
+  private CompositePost parseCompositePost(Element el) throws InvalidXMLException {
     String id = el.getAttributeValue("id");
+    if (id == null) id = PostDefinition.makeDefaultId(null, postIdSerial);
+
+    boolean sequential = XMLUtils.parseBoolean(el.getAttribute("sequential"), false);
+
+    ImmutableList.Builder<SinglePost> chBuilder = ImmutableList.builder();
+    for (Element child : el.getChildren("post")) {
+      chBuilder.add(parseSinglePost(new InheritingElement(child)));
+    }
+    ImmutableList<SinglePost> children = chBuilder.build();
+    // This should never happen, since it should've parsed single post instead.
+    if (children.isEmpty())
+      throw new InvalidXMLException("Expected children posts, but found none", el);
+
+    SinglePost fallback = null;
+    Node fallbackAttr = Node.fromAttr(el, "fallback");
+    if (fallbackAttr != null) {
+      fallback = factory.getFeatures().get(fallbackAttr.getValue(), SinglePost.class);
+      if (fallback == null) {
+        throw new InvalidXMLException(
```

Comment:
Don't we have a more generic way to error when a feature definition is missing?

Reply:
Unsure, this is mostly a copy from how the flag searches for an inner post a few lines lower on: https://github.com/PGMDev/PGM/pull/984/files/1a5371e3417272a84a471042970a2db90720dc4f#diff-7aa0581d0bce176dc53a9eadaca6d5a541de970854921dbe6b302cd51f542807R195